practiceCourseGenerator/practiceCourseGenerator.py

```diff
@@ -0,0 +1,233 @@
+'''
+
+@author: Kyle Dove
+
+Purpose: Download users list from Google Drive. Generate CSV files in 
+directory. Zip directory to be used to create Canvas practice courses.
+
+DATE           Author               Description   
+===========    ===============      ============================================
+Mar 31 2015    Kyle Dove			TLUNIZIN-470
+									Created.
+
+'''
+
+import logging
+import time
+import csv
+import requests
+import json
+import os
+import zipfile
+from apiclient.discovery import build
+from httplib2 import Http
+from oauth2client import file, client, tools
+from apiclient import errors
+from apiclient import http
+
+def download_file(service, drive_file):
+    download_url = drive_file['exportLinks']['text/csv']
+    logger.info('DownloadUrl: ' + download_url)
+    if download_url:
+            resp, content = service._http.request(download_url)
+            if resp.status == 200:
+                    logger.info('Status: %s' % resp)
+                    title = drive_file.get('title')
+                    path = './input.csv'
+                    file = open(path, 'wb')
+                    file.write(content)
+            else:
+                    logger.info('An error occurred: %s' % resp)
+                    return None
+    else:
+            # The file doesn't have any content stored on Drive.
+            return None
+
+def zipdir(path, zip):
+    for root, dirs, files in os.walk(path):
+        for file in files:
+            zip.write(os.path.join(root, file))
+
+#setup log
+
+#create logger 'canvas_scraper'
+logger = logging.getLogger('canvasFileGenerator')
+logger.setLevel(logging.INFO)
+logdate = time.strftime("%Y%m%d%H%M")
+
+#create file handler
+fh = logging.FileHandler('canvasFileGeneratorLog_' + logdate + '.log')
+fh.setLevel(logging.INFO)
+
+#create console handler
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+
+#create formatter and add it to the handlers
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+fh.setFormatter(formatter)
+ch.setFormatter(formatter)
+
+#add the handlers to the logger
+logger.addHandler(fh)
+logger.addHandler(ch)
+
+logger.info('Script Initiated')
+
+#def getInputFile():
+lines = []
+userSets = []
+users = []
+finalUsers = []
+userStrings = []
+dataSets = []
+directoryCounter = 0
+errorCount = 0
+outputDirectory = './canvas_files'
+userHeader = 'user_id,login_id,password,first_name,last_name,email,status'
+courseHeader = 'course_id,short_name,long_name,account_id,term_id,status,start_date,end_date'
+enrollmentHeader = 'course_id,user_id,role,section_id,status,associated_user_id'
+userDictionaries = []
+
+#Read properties file
+with open('propertiesProd.json') as data_file:    
+    data = json.load(data_file)
+
+logger.debug('urlPrefix: ' + str(data['URL_PREFIX']))
+logger.debug('driveFile: ' + str(data['DRIVE_FILE']))
+
+urlPrefix = str(data['URL_PREFIX'])
+urlPost = '/profile?access_token='
+
+#Download exceptions file
```

Comment:
Typo - I believe this is a user list, not an exceptions file? Is this the user list?